kmeta/names.go

```diff
@@ -55,12 +56,20 @@ func ChildName(parent, suffix string) string {
 			if d := longest - len(ret); d > 0 {
 				ret += suffix[:d]
 			}
-			// If due to trimming above we're terminating the string with a `-`,
-			// remove it.
-			return strings.TrimRight(ret, "-")
+			return makeValidName(ret)
 		}
 		//nolint:gosec // No strong cryptography needed.
 		n = fmt.Sprintf("%s%x", parent[:head-len(suffix)], md5.Sum([]byte(parent)))
 	}
 	return n + suffix
 }
+
+// If due to trimming above we're terminating the string with a non-alphanumeric character,
+// remove it. Also make sure things are lowercase
+func makeValidName(n string) string {
+	match := regexp.MustCompile(`^[a-zA-Z0-9]*$`)
```

Comment:
Capital letters are still not allowed in valid names. (Sorry, I think I slipped this into another comment earlier.)

This also means you don't need `strings.ToLower`.